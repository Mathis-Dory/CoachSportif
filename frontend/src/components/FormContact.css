.title {
  position: static;
  left: -100px;
  top: 25px;
  bottom: 25px;
}

.box {
    position: absolute;
    margin-top: 270px;
    margin-bottom: 50px;
    left: 50%;
    transform: translate(-50%, -50%);
    box-shadow: 0 5px 8px 0 darkred, 0 7px 20px 0 rgba(0, 0, 0, 0.2);
    background-color: sandybrown;
    border-radius:  10px 10px 10px 10px;
    width: 500px;
    height: 525px;
    color: white;
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande',
    'Lucida Sans', Arial, sans-serif;
    font-weight: bold;
    padding-top: 35px;

}

.subtitles{

    font-size: 20px;
    top: 30px;
    bottom: 20px;
    color: white;
}

.entry{
    display: block;
    padding-left: 10px;
    outline: none;
    border-radius: 2px;
    height: 40px;
    width: 350px;
    border: none;
    color: #595959;
    font-size: 17px;
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande',
    'Lucida Sans', Arial, sans-serif;
    font-weight: bold;
    margin-top: 15px;
    margin-bottom: 15px;
}

.textbox{
    display: block;
    padding: 10px;
    outline: none;
    border-radius: 2px;
    height: 135px;
    width: 350px;
    border: none;
    color: #595959;
    font-size: 14px;
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande',
    'Lucida Sans', Arial, sans-serif;
    font-weight: bold;
    margin-top: 15px;
    margin-bottom: 15px;
    resize : none;
    text-align: justify-all;
}

.pack{
  padding-left: 16%;
}

.form-input-btn-contact{
  width: 80%;
  height: 50px;
  margin-top: 10px;
  margin-left: 10%;
  border-radius: 2px;
  background-color: #663300;
  outline: none;
  border: none;
  color: #fff;
  font-size: 20px;
  font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande',
    'Lucida Sans', Arial, sans-serif;
  font-weight: bold;
}

.form-input-btn-contact:hover{
  cursor: pointer;
  background-color: #242222;
  transition: all 0.4s ease-out;
}


