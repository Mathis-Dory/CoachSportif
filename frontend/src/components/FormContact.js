import './Form.css';
import React, {useState} from "react";


const FormContact = () => {

    const [email, setEmail] = useState('');
    const [subject, setSubject] = useState('');
    const [demande, setDemande] = useState('');

    const sendData = async (event) => {
        event.preventDefault();
        try {
            let donnees = await fetch('/api/contact', {
                method: 'POST',
                headers: {
                    'Content-type': 'application/json',
                    'accept': 'application/json'
                }, body: JSON.stringify({email, subject, demande})
            })
            console.log("ok");
            console.log(donnees.body);
            console.log(donnees.status);
           // window.location.replace("/");
        } catch (jsp) {
            console.log(jsp);
        }
        alert("Nous avons envoyé votre message, vous recevrez une réponse d'ici peu, merci!")
    }
            return(
                <div>
                    <h1 className='title'>FORMULAIRE DE CONTACT</h1>
                    <form className="form-content-contact" onSubmit={(event) => {sendData(event)}}>
                        <div className='form'>

                            <div className="form-inputs">
                            <label className='form-label'>ADRESSE E-MAIL : </label>

                            <input className='form-input-contact' type='email' required value={email}
                                   name='email' placeholder='Entrez votre adresse e-mail'
                                   onChange={(jsp) => setEmail(jsp.target.value)}
                            />
                            </div>

                            <div className="form-inputs">
                            <label className='form-label'>SUJET : </label>
                            <br></br>
                            <input className='form-input-contact' type='text' required value={subject}
                                   name='subject' placeholder='Sujet de la demande'
                                   onChange={(jsp) => setSubject(jsp.target.value)}
                            />
                            </div>

                            <div className='form-inputs'>
                            <label className='form-label'>DÉCRIVEZ VOTRE DEMANDE : </label>
                            <br></br>
                            <textarea placeholder="Ecrivez votre demande ici" className='textbox' required
                                      value={demande} name='demande' maxLength='300'
                                      onChange={(jsp) => setDemande(jsp.target.value)}>
                            </textarea>
                            </div>

                        <button type="submit" className='form-input-btn'>ENVOYER LA DEMANDE</button>
                        </div>
                        </form>
                </div>
            );
}

export default FormContact;