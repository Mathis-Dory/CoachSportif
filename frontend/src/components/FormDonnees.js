import React, {useEffect, useState} from 'react';
import {Link} from "react-router-dom";
import './FormDonnees.css';

import {Redirect} from 'react-router-dom';

const FormDonnees = () => {

    // Met le state des inputs

    const [alias, setAlias] = useState('');
    const [taille, setTaille] = useState(null)
    const [email, setEmail] = useState('');
    const [poids, setPoids] = useState(null)
    const [premium, setPremium] = useState(null)

    useEffect(() => {
        const id = localStorage.getItem('id')
        const apiUrl = `/api/utilisateurs/`;
        fetch(apiUrl + localStorage.getItem("id"), {
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
                'Authorization': 'Bearer ' + localStorage.getItem('access_token')
            }
        })
            .then((res) => res.json())
            .then((response) => {
                setAlias(response[id].alias)
                setEmail(response[id].email)
                setTaille(response[id].taille)
                setPoids(response[id].poids)
                setPremium(response[id].premium)
            });
    }, []);

    const handleClick = (e) => {
        e.preventDefault()
        console.log("Modification des données")

        fetch('/api/utilisateurs/' + localStorage.getItem("id"), {
            method: 'PUT',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
                'Authorization': 'Bearer ' + localStorage.getItem('access_token')
            },
            body: JSON.stringify({alias, email, taille, poids})
        }).then(() => {
            //window.location.replace("/objectifs")
        }).catch(() => {
            console.log("erreur")
        })
    }

    switch (premium) {
        case null:
            return (
                <div>
                    <div className="spinner-border" role="status">

                    </div>


                </div>
            )
            break;
        case true:
            return (

                <div>
                    <h1>VOS DONNÉES</h1>
                    <div className='form-content-don'>
                        <form className="form-don">

                            <div className='form-inputs-don'>

                                <label className='form-label-don'>ALIAS</label>
                                <input
                                    className='form-input-don'
                                    type='text'
                                    name='text'
                                    placeholder=''
                                    value={alias}
                                    onChange={(e) => setAlias(e.target.value)}

                                />
                            </div>
                            <div className='form-inputs-don'>
                                <label className='form-label-don'>TAILLE</label>
                                <input
                                    className='form-input-don'
                                    type='number'
                                    name='number'
                                    placeholder=''
                                    value={taille}
                                    onChange={(e) => setTaille(e.target.value)}
                                />
                            </div>
                            <div className="form-inputs-don">
                                <label className='form-label-don'>POIDS</label>
                                <input
                                    className='form-input-don'
                                    type='number'
                                    name='alias'
                                    placeholder=''
                                    value={poids}
                                    onChange={(e) => setPoids(e.target.value)}
                                />
                            </div>
                            <div className='form-inputs-don'>
                                <label className='form-label-don'>EMAIL</label>
                                <input
                                    className='form-input-don'
                                    type='email'
                                    name='email'
                                    placeholder=''
                                    value={email}
                                    onChange={(e) => setEmail(e.target.value)}
                                />
                            </div>

                            <button onClick={handleClick} className='form-input-btn-don'>MODIFIER</button>


                        </form>

                    </div>
                </div>
            );
            break;
        case false:
            return (

                <div>
                    <h1>VOS DONNÉES</h1>
                    <div className='form-content-don'>
                        <form className="form-don">

                            <div className='form-inputs-don'>

                                <label className='form-label-don'>ALIAS</label>
                                <input
                                    className='form-input-don'
                                    type='text'
                                    name='text'
                                    placeholder=''
                                    value={alias}
                                    onChange={(e) => setAlias(e.target.value)}

                                />
                            </div>
                            <div className='form-inputs-don'>
                                <label className='form-label-don'>TAILLE</label>
                                <input
                                    className='form-input-don'
                                    type='number'
                                    name='number'
                                    placeholder=''
                                    value={taille}
                                    onChange={(e) => setTaille(e.target.value)}
                                />
                            </div>
                            <div className="form-inputs-don">
                                <label className='form-label-don'>POIDS</label>
                                <input
                                    className='form-input-don'
                                    type='number'
                                    name='alias'
                                    placeholder=''
                                    value={poids}
                                    onChange={(e) => setPoids(e.target.value)}
                                />
                            </div>
                            <div className='form-inputs-don'>
                                <label className='form-label-don'>EMAIL</label>
                                <input
                                    className='form-input-don'
                                    type='email'
                                    name='email'
                                    placeholder=''
                                    value={email}
                                    onChange={(e) => setEmail(e.target.value)}
                                />
                            </div>

                            <button onClick={handleClick} className='form-input-btn-don'>MODIFIER</button>

                            <button className='form-input-btn-prem'><Link className='form-input-btn-prem' to='/premium'>DEVENIR PREMIUM</Link></button>

                        </form>

                    </div>
                </div>
            );

            break;
        default:
            break;

    }
}

export default FormDonnees;