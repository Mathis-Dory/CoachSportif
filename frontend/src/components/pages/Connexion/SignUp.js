import React from 'react';
import '../../../App.css';
import FormSignup from "../../FormSignUp"

export default function SignUp() {
    return (
        <>
            <FormSignup/>;
        </>

    );
}
