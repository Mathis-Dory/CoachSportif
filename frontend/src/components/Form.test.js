import React from "react";
import ReactDOM from "react-dom";
import FormSignup from "./FormSignUp";
import FormSignin from "./FormSignIn";
import FormContact from "./FormContact";
import FormDonnees from "./FormDonnees";
import { MemoryRouter } from "react-router-dom";
import TestRenderer from "react-test-renderer";
import SignUp from "./pages/SignUp";

test("render sans que ça plante", () => {
  const div = document.createElement("div");
  ReactDOM.render(<MemoryRouter><FormSignup /></MemoryRouter>, div);
});

test("render sans que ça plante", () => {
  const div = document.createElement("div");
  ReactDOM.render(<MemoryRouter><FormSignin /></MemoryRouter>, div);
});

test("render sans que ça plante", () => {
  const div = document.createElement("div");
  ReactDOM.render(<MemoryRouter><FormContact /></MemoryRouter>, div);
});

test("render sans que ça plante", () => {
  const div = document.createElement("div");
  ReactDOM.render(<MemoryRouter><FormDonnees /></MemoryRouter>, div);
});

it("renders a form", () => {
  const testRenderer = TestRenderer.create(<MemoryRouter><FormSignin /></MemoryRouter>);
  const testInstance = testRenderer.toJSON();
  expect(testInstance).toMatchSnapshot();
});