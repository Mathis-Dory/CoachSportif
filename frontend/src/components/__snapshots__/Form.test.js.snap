// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a form 1`] = `
<div>
  <div
    className="form-content-in"
  >
    <form
      className="form"
    >
      <div
        className="form-inputs"
      >
        <label
          className="form-label"
        >
          EMAIL
        </label>
        <input
          className="form-input"
          name="email"
          onChange={[Function]}
          placeholder="Entrez votre e-mail"
          type="email"
          value=""
        />
      </div>
      <div
        className="form-inputs"
      >
        <label
          className="form-label"
        >
          MOT DE PASSE
        </label>
        <input
          className="form-input"
          name="password"
          onChange={[Function]}
          placeholder="Entrez mot de passe"
          type="password"
          value=""
        />
      </div>
      <button
        className="form-input-btn"
        onClick={[Function]}
      >
        SE CONNECTER
      </button>
      <div
        className="connexion"
      >
        <p
          className="conect"
        >
          Vous n'avez pas de compte, alors 
          <a
            className="inscrire"
            href="/sign-up"
            onClick={[Function]}
            style={
              Object {
                "color": "white",
              }
            }
          >
            INSCRIVEZ VOUS
          </a>
        </p>
      </div>
    </form>
  </div>
</div>
`;
